primitives/npos-elections/src/lib.rs

```diff
@@ -144,9 +145,84 @@ pub type VoteWeight = u64;
 /// A type in which performing operations on vote weights are safe.
 pub type ExtendedBalance = u128;
 
-/// The score of an assignment. This can be computed from the support map via
-/// [`EvaluateSupport::evaluate`].
-pub type ElectionScore = [ExtendedBalance; 3];
+/// The score of an election. This is the main measure of an election's quality.
+#[derive(Clone, Copy, PartialEq, Eq, Encode, Decode, MaxEncodedLen, TypeInfo, Debug, Default)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct ElectionScore {
+	/// The minimal winner, in terms of total backing stake.
+	///
+	/// This parameter should be maximized.
+	pub minimal_stake: ExtendedBalance,
+	/// The sum of the total backing of all winners.
+	///
+	/// This parameter should maximized
+	pub sum_stake: ExtendedBalance,
+	/// The sum squared of the total backing of all winners, aka. the variance.
+	///
+	/// Ths parameter should be minimized.
+	pub sum_stake_squared: ExtendedBalance,
+}
+
+impl ElectionScore {
+	fn iter(&self) -> impl Iterator<Item = ExtendedBalance> {
+		vec![self.minimal_stake, self.sum_stake, self.sum_stake_squared].into_iter()
+	}
+
+	/// Compares two sets of election scores based on desirability, returning true if `self` is
+	/// strictly `threshold` better than `other`.
+	///
+	/// Evaluation is done in a lexicographic manner, and if each element of `self` is `self *
+	/// epsilon` greater or less than `other`.
```

Comment:
```suggestion
	/// Evaluation is done in a lexicographic manner, and if each element of `self` is `other_element *
	/// epsilon` greater or less than `self_element`.
```
I think self * epsilon is wrong, because we do self.zip(other) which results in [(self_ele, other_ele), (self_ele, other_ele)]

So this line
```
.map(|(this, that)| (this.ge(&that), this.tcmp(&that, threshold.mul_ceil(that))))
```
is basically

```
.map(|(self_ele, other_ele)| (self_ele.ge(&other_ele), self_ele.tcmp(&that, threshold.mul_ceil(other_ele))))
```

Reply:
sorry can you elaborate a bit more? I re-read the code and I think it is as compact as it can be, and correct.

Reply:
The threshold is `other_element * epsilon` not `self_element * epsilon`. But maybe I am misunderstanding what you mean by `self * epsilon`